### PR TITLE
Attach client x509.Certificate to Batch if available

### PR DESCRIPTION
**Pull request by rob-stripe**
r? @adunham-stripe

### CODE REVIEW

File: server/v2/reader.go [hunk 1]
```diff
@@ -81,7 +83,29 @@ func (r *reader) ReadBatch() (*lj.Batch, error) {
...
+	// If the connection is over TLS and the client presented a client TLS
+	// cert, attach the cert to the Batch.
+	//
+	// Note that this does NOT check whether the TLS cert is valid - your
+	// TLS connection will need to have `ClientAuth` set to `tls.RequireAndVerifyClientCert`
+	// in order for certs to be verified.
+	var clientX509Cert *x509.Certificate
+	switch r.conn.(type) {
+	case *tls.Conn:
+		tlsClient := r.conn.(*tls.Conn)
```
Comment (adunham-stripe) [inline server/v2/reader.go]: Helpful go-ism

```suggestion
	switch tlsClient := r.conn.(type) {
	case *tls.Conn:
		// tlsClient is now of type *tls.Conn
```

**Review by adunham-stripe — CHANGES_REQUESTED** [review]
One small change that I didn't catch earlier - sorry! But otherwise LGTM.

File: server/v2/reader.go (hunk 1 above)
Comment (adunham-stripe) [reply server/v2/reader.go]: (docs [here](https://tour.golang.org/methods/16) if you're interested)

**Comment by rob-stripe** [comment]
Thanks - changed and tested! https://github.com/stripe/go-lumber/pull/2/commits/bdff6a5893f89c58067adcdb34abe350cd94818f

re-r? @adunham-stripe

**Review by adunham-stripe — APPROVED** [review]
Thank you!

**Pull request merged**